### PR TITLE
Add some CompareAttributeTests

### PR DESCRIPTION
**Pull request by hughbe**
And cleanup existing tests

There are some bugs (e.g. not checking that the property has index parameters, and not checking if the property can be read)

### CODE REVIEW

**Comment by hughbe** [comment]
Test Innerloop CentOS7.1 Release Build and Test

File: src/System.ComponentModel.Annotations/tests/CompareAttributeTests.cs [hunk 1]
```diff
@@ -2,100 +2,98 @@
...
-            // cannot check error message - not defined on ret builds
+            string newErrorMessage = attribute.FormatErrorMessage("name");
+            if (otherPropertyDisplayName == null || otherProperty.Equals(otherPropertyDisplayName))
+            {
+                Assert.Equal(previousErrorMessage, newErrorMessage);
+            }
+            else
+            {
+                Assert.NotEqual(previousErrorMessage, newErrorMessage);
+            }
```
Comment (stephentoub) [inline src/System.ComponentModel.Annotations/tests/CompareAttributeTests.cs]: I'm a bit confused by this test.  Can you explain what it's doing?  Maybe split it up into different tests?  Apparently something is causing the result of FormatErrorMessage to potentially change, but it's not clear what (calling Validate?  Accessing OtherPropertyDisplayName?).  And none of that seems to jive with the name of the test, Validate_Invalid_Throws.

**Comment by stephentoub** [comment]
One comment/question, otherwise LGTM. Thanks.

File: src/System.ComponentModel.Annotations/tests/CompareAttributeTests.cs (hunk 1 above)
Comment (hughbe) [inline src/System.ComponentModel.Annotations/tests/CompareAttributeTests.cs]: You're right.
The test was checking that we update the method `FormatErrorMessage` if we have an exception with a method that has a custom display name. This covers a branch in the `FormatErrorMessage` method.

probably easiest to understand by looking at the new test in the PR feedback commit!

**Comment by stephentoub** [comment]
Thanks, Hugh.

**Pull request merged**